Content.Server/Supermatter/Components/SupermatterComponent.cs

```diff
@@ -0,0 +1,166 @@
+using Content.Shared.Atmos;
+using Content.Shared.Sound;
+using Robust.Shared.GameObjects;
+using Robust.Shared.ViewVariables;
+using Robust.Shared.Analyzers;
+using Content.Server.Supermatter.EntitySystems;
+using Content.Shared.Whitelist;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+
+namespace Content.Server.Supermatter.Components
+{
+    [RegisterComponent, Friend(typeof(SupermatterSystem))]
+    public class SupermatterComponent : Component
+    {
+        public override string Name => "Supermatter";
+
+        //TODO: clean all this up more
+        //i've yet to see another component need so many variables and im not even using most of them
+        //the gas constants like HeatPenalty are supermatter specific and shouldnt be used elsewhere
+
+        [DataField("whitelist")]
+        public EntityWhitelist Whitelist = new();
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float Power {get; set;} = 0;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public Atmos.GasMixture? Mix {get; set;}
+
+        [ViewVariables(VVAccess.ReadOnly)]
+        public float GasmixPowerRatio {get; set;} = 0;
+
+        public float DynamicHeatModifier {get; set;} = 0;
+
+        public float PowerTransmissionBonus {get; set;} = 0;
+
+        public float PowerlossDynamicScaling {get; set;} = 0;
+
+        public float DynamicHeatResistance {get; set;} = 0;
+
+        public float PowerlossInhibitor {get; set;} = 0;
```

Comment:
you dont need to do `{get; set;]` just do `public float VariableName => 0` and it'll make it an autoproperty